pkg/registry/generic/etcd/etcd.go

```diff
@@ -459,16 +462,56 @@ func (e *Etcd) DeleteCollection(ctx api.Context, options *api.DeleteOptions, lis
 	if err != nil {
 		return nil, err
 	}
-	for _, item := range items {
-		accessor, err := meta.Accessor(item)
-		if err != nil {
-			return nil, err
-		}
-		if _, err := e.Delete(ctx, accessor.GetName(), options); err != nil {
-			return nil, err
+	// Spawn a number of goroutines, so that we can issue requests to etcd
+	// in parallel to speed up deletion.
+	workersNumber := registryoptions.GetDeleteCollectionParallelism()
```

Comment:
Please always pass these values in to the generic etcd as parameters, not as globals.  Globals are bad, we try to avoid them at all possible in the code base, and they break encapsulation and testing.